source/concurrent/Promise.ooc

```diff
@@ -109,7 +109,7 @@ Future: abstract class <T> {
 }
 
 _ThreadFuture: class <T> extends Future<T> {
-	_result: Object
+	_result: __onheap__ T
```

Comment:
Nice, I hadn't seen that. I would certainly prefer to not have to manually add this keyword, but I will need to look into it some more. @thomasfanell and @fredrikbryntesson , thoughts?